pytorch_lightning/trainer/trainer.py

```diff
@@ -2022,7 +2022,8 @@ def num_processes(self) -> int:
 
     @property
     def root_gpu(self) -> Optional[int]:
-        return self._accelerator_connector.root_gpu
+        rank_zero_deprecation("Method `trainer.root_gpu` is deprecated please use `trainer.strategy.root_device`")
```

Comment:
Since this is in a `@property`, we need to increase the stacklevel

```suggestion
        rank_zero_deprecation("Method `trainer.root_gpu` is deprecated please use `trainer.strategy.root_device`", stacklevel=5)
```